lightning/src/ln/reorg_tests.rs

```diff
@@ -102,16 +102,19 @@ fn do_test_onchain_htlc_reorg(local_commitment: bool, claim: bool) {
 
 		// Give node 1 node 2's commitment transaction and get its response (timing the HTLC out)
 		mine_transaction(&nodes[1], &node_2_commitment_txn[0]);
+		connect_blocks(&nodes[1], TEST_FINAL_CLTV - 1); // Confirm blocks until the HTLC expires
 		let node_1_commitment_txn = nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap();
 		assert_eq!(node_1_commitment_txn.len(), 3); // ChannelMonitor: 1 offered HTLC-Timeout, ChannelManger: 1 local commitment tx, 1 Offered HTLC-Timeout
-		assert_eq!(node_1_commitment_txn[1].output.len(), 2); // to-local and Offered HTLC (to-remote is dust)
-		check_spends!(node_1_commitment_txn[1], chan_2.3);
-		check_spends!(node_1_commitment_txn[2], node_1_commitment_txn[1]);
-		check_spends!(node_1_commitment_txn[0], node_2_commitment_txn[0]);
+		assert_eq!(node_1_commitment_txn[0].output.len(), 2); // to-local and Offered HTLC (to-remote is dust)
+		check_spends!(node_1_commitment_txn[0], chan_2.3);
+		check_spends!(node_1_commitment_txn[1], node_1_commitment_txn[0]);
+		check_spends!(node_1_commitment_txn[2], node_2_commitment_txn[0]);
 
 		// Confirm node 2's commitment txn (and node 1's HTLC-Timeout) on node 1
 		header.prev_blockhash = nodes[1].best_block_hash();
-		connect_block(&nodes[1], &Block { header, txdata: vec![node_2_commitment_txn[0].clone(), node_1_commitment_txn[0].clone()] });
+		let block = Block { header, txdata: vec![node_2_commitment_txn[0].clone(), node_1_commitment_txn[2].clone()] };
+		std::mem::drop(node_1_commitment_txn);
```

Comment:
Consider just cloning the entire `Vec` rather than holding a lock, given you're gonna clone some elements anyway.